### DISCUSSION TITLE
Music playlist

### DISCUSSION THREAD

**Discussion by Danielt561** [OpenSageTV/Community#147] [FAQ]
Is there a way to import a music playlist into Sage

**Comment by wayner9** [comment]
Hardly anyone uses SageTV for music and I don't think the music part of SageTV ever got much attention.

**Reply by Bionicmerlin** [reply]
I used to use it for music before the 64 bit version came in.  I ran the Winamp plugin and used Milkdrop.

**Comment by Danielt561** [comment]
Thank you for your response! I guess Sage is just going to fade away. I have all my music saved on my computer and it is more convenient to use the remote control to navigate than the mouse and keyboard with another program.

**Comment by wayner9** [comment]
SageTV has been on live support since Google bought it in 2011.  There was a bit of a resurrection when SageTV was open sourced, but that was short lived and almost all of the open source developers have faded away in the last few years.

But music development died much earlier.

For music there are dozens of options, you are far better off than for recorded TV.  Almost everyone has gone to streaming for music and uses Spotify or Apple Music.  You can still use your own files with those services.  And there are apps that are maintained that let you use your own music - I am pretty sure that you can do this in Plex.  But you would need a different end device than a SageTV extender.  But most of the devices do not have audio outputs, other than HDMI which has both audio and video.

**Reply by Sagetvmaster** [reply]
Do you think Narflex and his tight group of longtime admins are quietly using an updated version which can't be made public?  I'd like to think so.  Wish I was in that small group :)
It's far to awesome software to let die, but alas, that's likely what will happen.  They sold to Google in 2010, and yet here I am, still using it with multiple sources, file formats, dvd rips, etc.  Not to mention the media extenders, which are fine if used within their 2010 codec limitations.

**Reply by wayner9** [reply]
I would doubt that very much.  Linear TV is dying a slow death.  And the DVR system from your cable company, like the Comcast Xfinity (aka Rogers Ignite here in Canada) is now very good and has features that SageTV does not.  The SageTV extenders are still pretty good but they are about 15 years old are also pretty obsolete.  Other than live sports I see little reason for needing a DVR since most other options are superior for watching TV shows, especially if you know how to use BitTorrent and other tools like that.  

And other platforms like Plex and Jellyfin are now much better than SageTV for playing back recorded or ripped movies or TV shows.  And they have clients for pretty much any modern playback device like iOS, iPadOS, Android, Roku, FireTV, AppleTV, AndroidTV, Samsung, Xbox, etc.

So there is WAY less need for SageTV today then there was 5, 10 or 15 years ago.  And even back then a PC based DVR was always a niche selection for geeks.  Microsoft tried this and realized it didn't work with Windows MCE.

If I could get reasonably priced STBs from my cable company I would shut down SageTV for new recordings and just use the cable company boxes. I am moving in that direction right now as I am coming up with other solutions like HDMI over CAT6.  I have one impediment - an autistic daughter who knows how to use SageTV and she gets very upset when anything changes, even if she just has to use the SageTV backup server.

**Comment by Sagetvmaster** [comment]
Too bad.  We love linear TV.  Hate streaming, passwords, internet outages, invasive data breaches, etc.  Give me a set-top box, and cable subscription, some TiVo's, some OTA antennas and a satellite subscription and we'll be happy.  I imagine content providers would have ended up suing SageTV to death over it's ability to record, etc.  Bummer
________________________________

**Reply by wayner9** [reply]
Cable is now usually delivered via internet with IPTV services like Comcast Xfinity so if your internet goes down it is very likely that your cable goes down as well. I had a TiVo too back in the old days, but the cable DVR services are pretty good and they are cloud based.  That has pros and cons but the pro is that you can watch from anywhere, especially if you have a VPN. 

There was no reason to ever sue SageTV as it was just using existing technologies and devices to record TV content.  SageTV used devices from Silicon Dust (HD Home Run), Hauppauge, etc.  It did not circumvent any copy protection, so it is unlikely it would be sued.

OTA may actually be copy protected in the future since the ATSC 3.0 spec has that as part of its spec and you may need to pay to see OTA content.

**Comment by northwestview** [comment]
Sorry no one answered your question and all have hijacked the thread with opinions about Sage's longevity.
To answer your question, if you put an .m3u playlist in the directory that Sage searches for music files in, it will import the playlist on its next scan of your library (or you can initiate a manual scan).

**Reply by Danielt561** [reply]
Thank you for answering my question. Do you know what program I can use to create the list in that format or someway to convert the playlists I already have?

**Reply by northwestview** [reply]
Try one of these:  https://www.techradar.com/news/the-best-free-music-player

**Reply by Danielt561** [reply]
I saved the playlist in a M3U file, and it does not seem to import into Sage. Any other suggestions?

**Reply by Danielt561** [reply]
After several days it finally imported! I have made several playlists, and they seem to import the same day, weird.

**Reply by northwestview** [reply]
Sorry, didn't see these as Github didn't mark the thread as having new messages.
As I mentioned, you can initiate a manual scan. Menu: Setup->Scan Imported Media.
You can set the frequency of Sage's automatic library rescan: Menu: Detailed Setup->General->Automatic Library Rescan Frequency
You can also view the SageTV startup log to see if it found your .m3u files and imported them correctly. Startup log is in C:\Program Files (x86)\SageTV\SageTV\sagetv_0.txt
Open it in notepad and search for your music directory name to find a line that looks like this:
Wed 8/28 21:38:23.713 [LibraryImporter@28c601] Starting to scan lib import root: <your music directory>
Sage will also post system messages if there are unresolved file paths in your m3u: Menu: Setup->System Messages

**Reply by northwestview** [reply]
Should have been 
You can set the frequency of Sage's automatic library rescan: Menu: Setup->Detailed Setup->General->Automatic Library Rescan Frequency

**Comment by CptKirk-1701** [comment]
Google search: .m3u playlist creator

Many out there.
